appcelerator/titanium_mobile_windows
# [TIMOB-26019] Support x64 build target

- [TIMOB-26019](https://jira.appcelerator.org/browse/TIMOB-26019)
- [TIMOB-25999](https://jira.appcelerator.org/browse/TIMOB-25999)

### Building x64 target using `--architecture x64` option.

```
appc run -p windows --target ws-local -l trace --architecture x64 --build-only
```

How to test:

1. Build app with `--architecture x64` option
2. Open `C:\Users\USERNAME\.titanium\vsbuild\PROJECT_NAME\win10.x64\PROJECT_NAME.sln`
3. Visual Studio should target `x64` platform (See screenshot below)

![vs](https://user-images.githubusercontent.com/1661068/40344360-696ae0f2-5dcf-11e8-91e0-57e777163fca.png)

### Building ARM target using `--architecture arm` option.

```
appc run -p windows --target ws-local -l trace --architecture arm --build-only
```

How to test:

1. Build app with `--architecture arm` option
2. Open `C:\Users\USERNAME\.titanium\vsbuild\PROJECT_NAME\win10.ARM\PROJECT_NAME.sln`
3. Visual Studio should target `ARM` platform

### x64 module support

How to test:

1. Create new module with `appc new`.
2. Build the module (`cd MODULE_PROJECT; cd windows;  appc ti build -p windows --build-only -l trace`
3. Use the module in `x64` project
